graviti/portex/builder.py

```diff
@@ -238,18 +239,29 @@ def build(self) -> Type["PortexExternalType"]:
 
         params.add(Param("nullable", False, ptype=PTYPE.Boolean))
 
-        package = self._builder.package
-
-        type_: Type[PortexExternalType] = type(
-            self._name,
-            (PortexExternalType,),
-            {"params": params, "factory": factory, "package": package},
-        )
-
-        package[self._name] = type_
+        type_ = self._get_external_type(params, factory)
+        self._builder.package[self._name] = type_
 
         return type_
 
+    def _get_external_type(
+        self, params: Params, factory: Union[TypeFactory, VariableFactory]
+    ) -> Type[PortexExternalType]:
+        parent_types: List[Type["PortexType"]] = [PortexExternalType]
+        class_attrs: Dict[str, Any] = {
+            "params": params,
+            "factory": factory,
+            "package": self._builder.package,
+        }
+
+        if isinstance(factory, TypeFactory):
```

Comment:
I think factory should always be `TypeFactory` here, under what situation the factory will not be a `TypeFactory`?